src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -120,6 +120,7 @@ internal static MsalTokenResponse CreateFromiOSBrokerResponse(Dictionary<string,
                     AccountUserId = homeAcctId != null ? AccountId.ParseFromString(homeAcctId).ObjectId : null,
                     TenantId = homeAcctId != null ?  AccountId.ParseFromString(homeAcctId).TenantId : null,
                     Upn = (metadataDictionary?.ContainsKey(TokenResponseClaim.Upn) ?? false) ? metadataDictionary[TokenResponseClaim.Upn] : null,
+                    CorrelationId = responseDictionary[BrokerResponseConst.CorrelationId],
```

Comment:
Are you sure it's always there? Maybe it's worth doing a `ContainsKey` check?

Reply:
Fixed with other PR. https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3399/